enclave-runtime/src/rpc/author/author_tests.rs

```diff
@@ -18,50 +18,128 @@
 
-	pub fn submitting_to_author_inserts_in_pool() {
-		let top_pool = Arc::new(TrustedOperationPoolMock::default());
+	type TestAuthor<F> = Author<TrustedOperationPoolMock, F, HandleStateMock, ShieldingCryptoMock>;
 
-		let shard_id = shard_id();
-		let mut state_facade = HandleStateMock::default();
-		state_facade.init_shard(&shard_id).unwrap();
+	pub fn top_encryption_works() {
+		// the sgx crypto crate lacks unit tests, one of the reasons being that the
+		// crypto structs are only available in SGX mode and cannot be tested using cargo test
+		// so we test some of the functionality here, where we encrypt and decrypt trusted operations
+		// using a RSA3072 key
 
-		let encryption_key = ShieldingCryptoMock::default();
+		let trusted_call = TrustedOperation::from(trusted_call_signed());
+		let trusted_getter = TrustedOperation::from(trusted_getter_signed());
+
+		assert_eq!(trusted_call, encrypt_and_decrypt_top(&trusted_call));
+		assert_eq!(trusted_getter, encrypt_and_decrypt_top(&trusted_getter));
+	}
+
+	fn encrypt_and_decrypt_top(top: &TrustedOperation) -> TrustedOperation {
+		let encryption_key = Rsa3072KeyPair::new().unwrap();
+		let encrypted_top = encryption_key.encrypt(top.encode().as_slice()).unwrap();
+		let decrypted_top = encryption_key.decrypt(encrypted_top.as_slice()).unwrap();
 
-		let author = Author::new(top_pool.clone(), Arc::new(state_facade), encryption_key.clone());
+		TrustedOperation::decode(&mut decrypted_top.as_slice()).unwrap()
+	}
+
+	pub fn submitting_to_author_inserts_in_pool() {
+		let (author, top_pool) = create_author_with_filter(AllowAllTopsFilter);
 		let top = TrustedOperation::from(trusted_getter_signed());
-		let top_encrypted = encryption_key.encrypt(top.encode().as_slice()).unwrap();
 
-		let submit_future = async { author.submit_top(top_encrypted, shard_id).await };
-		let submit_response: H256 = executor::block_on(submit_future).unwrap();
+		let submit_response: H256 = submit_top(author, &top).unwrap();
 
 		assert!(!submit_response.is_zero());
 
 		let submitted_transactions = top_pool.get_last_submitted_transactions();
 		assert_eq!(1, submitted_transactions.len());
 	}
 
+	pub fn submitting_call_to_author_when_top_is_filtered_returns_error() {
+		let (author, top_pool) = create_author_with_filter(GettersOnlyFilter);
+		let top = TrustedOperation::from(trusted_call_signed());
+
+		let submit_response = submit_top(author, &top);
+
+		assert!(submit_response.is_err());
+		assert!(top_pool.get_last_submitted_transactions().is_empty());
+	}
+
+	pub fn submitting_getter_to_author_when_top_is_filtered_inserts_in_pool() {
+		let (author, top_pool) = create_author_with_filter(GettersOnlyFilter);
+		let top = TrustedOperation::from(trusted_getter_signed());
+
+		let submit_response = submit_top(author, &top).unwrap();
+
+		assert!(!submit_response.is_zero());
+		assert_eq!(1, top_pool.get_last_submitted_transactions().len());
+	}
+
+	fn submit_top<F: Filter<Value = TrustedOperation>>(
```

Comment:
We're now creating a lot of duplicate test code: https://github.com/integritee-network/worker/blob/master/enclave-runtime/src/tests.rs#L535-L548

Might be worth it to somehow join these.

Reply:
I'm not sure I understand what you're saying here - the tests have some wrapper code to call `submit_top`. That wrapper code is shared for all tests, no? (as you're code snippet shows).

Reply:
Doesn't need to be in this PR. We could create a "clean up" issue for tests.